### ISSUE TITLE
Deletion of menu items on the Navigation screen.

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#21282]
**Is your feature request related to a problem? Please describe.**
#21036 implemented the experimental nav screen. Currently menu items can be added or modified, but can't be removed. 

**Describe the solution you'd like**
In addition to `get` and `save` functions, the core data package also exposes a `remove`/`delete` function for entities. This is used to allow the deletion of menu items in this code:
https://github.com/WordPress/gutenberg/blob/37bda8a5c37c84894be11fa93a75d63f8b26974b/packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js#L93

**Comment by draganescu** [comment]
This is a bit complicated by the fact that the menu items which the `useNavigationBlocks` hook gets and serves up to it's calee's are not connected by any value to the menu items in the DB (ideally this should be by ID).

One reason for this is that in the navigation link block the concept of ID is completely different and refers to the id of the post linked to. In the context of menu items ID is the menu item's entry number in the `wp_postmeta`.

Since `useNavigationBlocks` artificially creates `innerBlocks` out of menu items we could:

1) reuse the id attribute of `navigationLink` in this context to mean something else
2) add a new attribute to `navigationLink` that is only used in this context

I try to use (1) since the two contexts are not likely to be interchangeable.

**Comment by draganescu** [comment]
Eventually, the problem we hit here is about the entities system not supporting `delete` operations on `Entity` data. We went ahead and tried different approaches to implement delete (with and without undo) but we have uncovered something which appears to be a bug in the Entities system, explained briefly like this:

![image](https://user-images.githubusercontent.com/107534/82032326-b5abeb00-96a3-11ea-8f55-3fe4157e1168.png)

There is something that calls `receiveEntityRecords` after the `delete` flow has completed. This causes that in effect:

1. The items get deleted from the DB by the `apiFetch` call
2. The items get removed from the state by the `REMOVE_ITEMS` action in the `items` reducer
3. The items appear back in the state by some mysterious call to `receiveEntityRecords`

Step 3 is done through some auto-magic wiring between `apiFetch` or `getEntityRecords` and the state. It is weird and unexpected.
